src/main/java/com/minecraftabnormals/buzzier_bees/core/other/BBCompat.java

```diff
@@ -61,6 +120,33 @@ public static void registerDispenserBehaviors() {
 		DispenserBlock.registerDispenseBehavior(BBItems.BOTTLE_OF_BEE.get(), new BeeBottleDispenseBehavior());
 		DispenserBlock.registerDispenseBehavior(BBItems.BOTTLE_OF_SILVERFISH.get(), new BugBottleDispenseBehavior());
 		DispenserBlock.registerDispenseBehavior(BBItems.BOTTLE_OF_ENDERMITE.get(), new BugBottleDispenseBehavior());
+
+		DataUtil.registerAlternativeDispenseBehavior(Items.GLASS_BOTTLE, (source, stack) -> !BlockUtil.getEntitiesAtOffsetPos(source, CreatureEntity.class, e -> ENTITY_TYPE_TO_BOTTLE_MAP.containsKey(e.getType())).isEmpty(), new DefaultDispenseItemBehavior() {
+			@Override
+			protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
+				Entity entity = BlockUtil.getEntitiesAtOffsetPos(source, Entity.class, e -> ENTITY_TYPE_TO_BOTTLE_MAP.containsKey(e.getType())).get(0);
+				ItemStack bottled = ENTITY_TYPE_TO_BOTTLE_MAP.get(entity.getType()).apply(entity);
+				stack.shrink(1);
+				if (stack.isEmpty()) {
+					stack = bottled.copy();
+				} else if (source.<DispenserTileEntity>getBlockTileEntity().addItemStack(bottled.copy()) < 0) {
+					new DefaultDispenseItemBehavior().dispense(source, bottled.copy());
```

Comment:
Not really a fan of this initialization here to do one task.

Reply:
Don't know exactly what you mean, there's no point just returning there as the sounds need to be played.

Reply:
> as the sounds need to be played

Then do that. Initialization to do a simple task isn't wise.

Reply:
Uhh... are you sure initialisation is expensive? Duplicating code would look really ugly